### DISCUSSION TITLE
Any idea that is we can get also the lyrics from the spotify api

### DISCUSSION THREAD

**Discussion by mmw1984** [navidrome/navidrome#2703] [Ideas]
I am too lazy to check if it is possible.

**Comment by metalheim** [comment]
Lyrics are usually subject to copyright, so usually lyrics APIs (at least the good ones) are pay-per-use.

There are projects that use Spotify to retreive lyrics, but they dont use the official API, but spotifys internal, undocumented API. That API is not designed nor documented for public use, and is additionally not licensed to be used by anyone but Spotify itself.
Using undocumented APIs is a maintanance nightmare (it can change at any moment that spotify desires) and usage is also of questionable legality.

The only decent free Lyrics API i know of is Genius Lyrics

**Reply by Yehuda-Bergstein** [reply]
Do you think adding this to Navidrome (using Genius api to display lyrics without changing the file itself) is possible?

I tried looking for this in the discord etc couldn't find this idea mentioned anywhere

**Reply by metalheim** [reply]
Digging into the Genius API https://docs.genius.com/ it turns out that it returns everything about a song **but not** the lyrics itself.

An open source project that fetches Genius Lyrics https://github.com/farshed/genius-lyrics-api does actually only use the API to find the song (Genius songId) and then scrapes the lyrics from the html of Genius' website.
So the **Lyrics aren't actually available via API**. Probably for the same reasons i stated earlier.

So your only option is to fetch the lyrics **somehow** and embed them into the files. In the future, navidrome may utilize `.lrc` files aswell.

**Reply by deluan** [reply]
> So your only option is to fetch the lyrics somehow and embed them into the files. In the future, navidrome may utilize .lrc files aswell.

Yes, that will be implemented at some point after we merge the #2656 PR.
